#!/usr/bin/env python
# DWA Planner configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t
gen = ParameterGenerator()

gen.add("min_linear_velocity", double_t, 0, "The minimum desired linear velocity for the robot. [m/s]", 0.3, 0, 1.5)
gen.add("nominal_linear_velocity", double_t, 0, "The nominal desired linear velocity for the robot. [m/s]", 1.0, 0, 1.5)
gen.add("min_range", double_t, 0, "The minimum range at which shadows effect the robot", 1.25, 0, 5.0)
gen.add("nominal_range_min", double_t, 0, "The minimum range at which shadows cause the robot to go the nominal velocity. [m]", 2.5, 0, 5.0)
gen.add("nominal_range_max", double_t, 0, "The maximum range range at which shadows cause the robot to go the nominal velocity. [m]", 3.0, 0, 5.0)
gen.add("max_range", double_t, 0, "The maximum range that shadows effect the robot", 3.25, 0, 5.0)
gen.add("forward_offset", double_t, 0, "The position forwards from the robot origin to measure distances from. [m]", 0.5, -1.5, 1.5)
gen.add("half_angle", double_t, 0, "The angle points are considere in front of the robot. [m]", 1.57, 0, 3.157)
gen.add("shadow_threshold", double_t, 0, "The percentage of points needed to be in the half angle to be turn on the limiter. [m]", 0.75, 0.0, 1.0)

exit(gen.generate("base_local_planner", "base_local_planner", "ShadowSpeedLimiter"))
